nixos/modules/virtualisation/nixos-containers.nix

```
@@ -4,6 +4,13 @@ with lib;
 
 let
 
+  configurationEtc = if versionAtLeast config.system.stateVersion "20.09" then "nixos-containers" else "containers";
+  configurationDirectory = "/etc/${configurationEtc}";
+  stateDirectory = (
+    if versionAtLeast config.system.stateVersion "20.09" then "/var/lib/nixos-containers"
+    else "/var/lib/containers"
+  );
```

Maybe something like this would work?

```suggestion
configurationPrefix = optionalString (versionAtLeast config.system.stateVersion "20.09") "nixos-";
configurationDirectory = "/etc/${configurationPrefix}containers";
stateDirectory = "/var/lib/${configurationPrefix}containers";
```

Though I guess it's less explicit. Just something I thought of while looking at this.

---

I like it :) Thanks!